File: monegros-restaurant-dianamonegrodiaz/src/monegros/restaurant/entities/StockProduct.java
package monegros.restaurant.entities;

import java.util.Objects;

public class StockProduct {
    private Long id;
    private Stock stock;
    private Product product;
    private Integer quantity;
    private MeasureUnit measureUnit;

    public StockProduct() {
    }

    public StockProduct(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Stock getStock() {
        return stock;
    }

    public void setStock(Stock stock) {
        this.stock = stock;
    }

    public Product getProduct() {
        return product;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public MeasureUnit getMeasureUnit() {
        return measureUnit;
    }

    public void setMeasureUnit(MeasureUnit measureUnit) {
        this.measureUnit = measureUnit;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        StockProduct that = (StockProduct) o;
        return Objects.equals(id, that.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: jmillord-accelerated-restaurant/src/entities/StockProduct.ts
import { Stock } from './Stock'
import { Product } from './Product'
import { MeasureUnit } from './MeasureUnit'


import { Md5 } from 'ts-md5/dist/md5'

export class StockProduct {
  private id: string;
  private stock: Stock;
  private product: Product
  private quantity: number;
  private measureUnit: MeasureUnit

  constructor() { }


  public getId(): string {
    return this.id;
  }

  public setId(id: string): void {
    this.id = id;
  }

  public getStock(): Stock {
    return this.stock;
  }

  public setStock(stock: Stock): void {
    this.stock = stock;
  }

  public getProduct(): Product {
    return this.product;
  }

  public setProduct(product: Product): void {
    this.product = product;
  }

  public getQuantity(): number {
    return this.quantity;
  }

  public setQuantity(quantity: number): void {
    this.quantity = quantity;
  }

  public getMeasureUnit(): MeasureUnit {
    return this.measureUnit;
  }

  public setMeasureUnit(measureUnit: MeasureUnit): void {
    this.measureUnit = measureUnit;
  }

  public hashCode() {
    return Md5.hashStr(this.id)
  }


}